### PR TITLE
*: Added server_debug to have a better experience when using a debugger

### PR DESCRIPTION
**Pull request by mjonss**
### What problem does this PR solve?

Added build option for debug build of tidb-server

Problem Summary:
When using a debugger as delve, it is hard to follow the execution and inspect some areas due to optimized build, this makes it easier to build a debug version of the server binary to run in a debugger.

### What is changed and how it works?

What's Changed: Added `server_debug` `make` target

How it Works: `make server_debug` which will create `bin/tidb-server-debug`

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)
- [X] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by bb7133 — APPROVED** [review]

**Comment by bb7133** [comment]
PTAL @blacktear23

**Comment by blacktear23** [comment]
@bb7133  LGTM

**Review by morgo — APPROVED** [review]
